api/service/stagedstreamsync/syncing.go

```diff
@@ -156,67 +166,119 @@ func getBlockDbPath(beacon bool, loopID int, dbDir string) string {
 		}
 	} else {
 		if loopID >= 0 {
-			return fmt.Sprintf("%s_%d", filepath.Join(dbDir, "cache/blocks_db"), loopID)
+			return fmt.Sprintf("%s_%d_%d", filepath.Join(dbDir, "cache/blocks_db"), shardID, loopID)
 		} else {
-			return filepath.Join(dbDir, "cache/blocks_db_main")
+			return fmt.Sprintf("%s_%d", filepath.Join(dbDir, "cache/blocks_db_main"), shardID)
 		}
 	}
 }
 
+func (s *StagedStreamSync) Debug(source string, msg interface{}) {
+	// only log the msg in debug mode
+	if !s.config.DebugMode {
+		return
+	}
+	pc, _, _, _ := runtime.Caller(1)
+	caller := runtime.FuncForPC(pc).Name()
+	callerParts := strings.Split(caller, "/")
+	if len(callerParts) > 0 {
+		caller = callerParts[len(callerParts)-1]
+	}
+	src := source
+	if src == "" {
+		src = "message"
+	}
+	// SSSD: STAGED STREAM SYNC DEBUG
+	if msg == nil {
+		fmt.Printf("[SSSD:%s] %s: nil or no error\n", caller, src)
```

Comment:
use utils.logger() ?

Reply:
This is the Debug function designed to print information in the console instead of the log file. As mentioned earlier, the Debug function prints each process very frequently, leading to an excessive amount of log lines. The underlying concept of the debug feature is to monitor data in real-time.